src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -382,6 +375,15 @@ private string GetRelativePath(RouteTemplate parsedTemplate)
                 }
             }
 
+            // Set the default status only when no status has already been set explicitly
+            if (objectTypes.Count == 0)
+            {
+                if (type != null && type != typeof(void))
+                {
+                    objectTypes[StatusCodes.Status200OK] = type;
+                }
+            }
```

Comment:
if an action returns `void`, then it would by default not have any apidescriptions...metadata needs to be explicitly decorated ex: https://github.com/aspnet/Mvc/blob/dev/test/WebSites/ApiExplorerWebSite/Controllers/ApiExplorerResponseTypeWithoutAttributeController.cs#L14